pkg/api/handlers/libpod/images.go

```diff
@@ -498,12 +505,61 @@ func CommitContainer(w http.ResponseWriter, r *http.Request) {
 	if len(query.Repo) > 0 {
 		destImage = fmt.Sprintf("%s:%s", query.Repo, tag)
 	}
-	commitImage, err := ctr.Commit(r.Context(), destImage, options)
+
+	var commitImage *libimage.Image
+
+	success := false
+	runCtx, cancel := context.WithCancel(r.Context())
+	go func() {
+		defer cancel()
+		commitImage, err = ctr.Commit(r.Context(), destImage, options)
+		if err == nil {
+			success = true
+		} else {
+			stdout.Write([]byte(err.Error() + "\n"))
```

Comment:
This seems to be the problem: merely printing the error is not the same as actually posting it through `utils.Error` as in line 524 -- I don't know how Go works, but I suspect that this `err` is only scoped inside this go func, and doesn't exist any more outside, so this construction short-circuits the error handling below.